internal/client/client.go

```diff
@@ -453,6 +454,40 @@ func (c *CSAPI) MustDoFunc(t *testing.T, method string, paths []string, opts ...
 	return res
 }
 
+// MustDoFuncUntil is the same as MustDoFunc but retries up to 5 times,
+// with exponential backoff (10 ms, 40 ms, 160 ms, ...) whilst the predicate is not satisfied.
+//
+// It will retry a successful call (2xx HTTP response code) until the predicate
+// described by `checkFunc` is satisfied, timing out after 5 retries.
+//
+// Caveats:
+// 1. You can't use MustDoFuncUntil to repeat a successful call until it fails.
+// 2. You can't use MustDoFuncUntil to repeat a failing call until it suceeds.
+func (c *CSAPI) MustDoFuncUntil(t *testing.T, method string, paths []string, checkFunc func(*http.Response) bool, opts ...RequestOpt) *http.Response {
```

Comment:
Please don't add new functions which are basically the same as other functions. Add a `RequestOpt`.